guesser.rb

```diff
@@ -0,0 +1,45 @@
+def guesser
+    loop do # Outer loop to restart game
+        puts "Think of a number between 1 and 100"
+        puts "I will guess. If I am too high, type H; if I am too low, type L."
+        puts "If I am correct, type Y."
+
+        low_num = 1
+        high_num = 100
+        guesses = []    # Array to hold guesses
```

Comment:
❕ This totally works, but consider that you can use the `high_num` and `low_num` to provide the same tracking without having to store all the guesses in an array: If `high_num` is ever less than `low_num`, someone is lying.

Let's look at this using a smaller range of 1-10 (ignore any scary `puts` string interpolation lines):

```ruby
low_num = 1
high_num = 10
puts "Okay, let me guess..."
loop do
  guess = (high_num + low_num)/2

  # output the possible numbers and the guess
  puts "\tlowest: #{low_num}\n\thighest: #{high_num}\n\tguess: #{guess}"

  # This simulates a (cheating) user response by returning a random boolean where `true` is "too high"
  too_high = [true, false].sample

  # output the "user response"
  puts "\tGuess is too #{too_high ? "🔺 high":"🔻 low"}!"

  if too_high
    high_num = guess - 1
  else
    low_num = guess + 1
  end

  if high_num < low_num
    puts "😤 Wait...you're lying to me!"
    puts "   You said #{guess} was too #{too_high ? "high":"low"} but the #{too_high ? "lowest" : "highest"} possible guess is #{too_high ? high_num + 1 : low_num - 1}"
    break
  end

  puts "\nI am going to guess again..."
end
```

Here is an output from this program:
```ruby
Okay, let me guess...
	lowest: 1
	highest: 10
	guess: 5
	Guess is too 🔺 high!

I am going to guess again...
	lowest: 1
	highest: 4
	guess: 2
	Guess is too 🔻 low!

I am going to guess again...
	lowest: 3
	highest: 4
	guess: 3
	Guess is too 🔺 high!
😤 Wait...you're lying to me!
      You said 3 was too high but the lowest possible guess is 3
```